day_3/story_game.py

```diff
@@ -1,9 +1,74 @@
-from scenario import SCENARIO
+from scenario import GREETING, SCENARIO
 
 
 class Node:
-    def __init__(self, position, left, right, text):
+    def __init__(self, position, left_pos, left, right_pos, right, text):
         self.position = position
+        self.left_event = left_pos
         self.left = left
+        self.right_event = right_pos
         self.right = right
         self.text = text
+
+
+class Story:
+    def __init__(self, story_dict: dict, greeting: str = ''):
+        self.story_dict = story_dict
+        self.greeting = greeting
+        self.story = {}
+
+    def make_story(self):
+        for position, details in self.story_dict.items():
+            node = Node(
+                position=position,
+                left_pos=details['left_pos'],
+                left=details['left'],
+                right_pos=details['right_pos'],
+                right=details['right'],
+                text=details['text'],
+            )
+            self.story[position] = node
+        event_codes = self.story.keys()
+        for event_code, event in self.story.items():
+            if event.left_event in event_codes:
+                event.left_event = self.story[event.left_event]
+            if event.right_event in event_codes:
+                event.right_event = self.story[event.right_event]
+
+    @staticmethod
+    def infinite_query(phrase, valid_answer):
+        answer = None
+        while answer not in valid_answer:
+            answer = input(phrase)
+        return answer
```

Comment:
_nit:_ just to imply that there should be multiple answers. The one might try to use it as `infinite_query('', 1)`, expecting that all answers but `1` should be ignored.
```suggestion
    def infinite_query(phrase, valid_answers):
        answer = None
        while answer not in valid_answers:
            answer = input(phrase)
        return answer
```